no method named `spi3_remap` found

error message:
```
   Compiling stm32f1xx-hal v0.7.0
error[E0599]: no method named `spi3_remap` found for mutable reference `&mut stm32f1::W<u32, Reg<u32, _MAPR>>` in the current scope
   --> C:\Users\xxx\scoop\persist\rustup\.cargo\registry\src\github.com-1ecc6299db9ec823\stm32f1xx-hal-0.7.0\src\spi.rs:226:35
    |
226 |         mapr.modify_mapr(|_, w| w.spi3_remap().bit(REMAP::REMAP));
    |                                   ^^^^^^^^^^ help: there is an associated function with a similar name: `spi1_remap`
```

Cargo.toml:
```toml
stm32f1xx-hal = { version = "0.7.0", features = ["rt", "stm32f103", "high"] }
```

target MCU:
STM32F103ZET6

I used new commit<0f367e6>, and could not compile stm32f1xx-hal too.
```
Checking stm32f1xx-hal v0.7.0 (https://github.com/stm32-rs/stm32f1xx-hal#0f367e60)
error[E0599]: no method named `spi3_remap` found for mutable reference `&mut stm32f1::W<u32, Reg<u32, _MAPR>>` in the current scope
   --> C:\Users\92307\.cargo\git\checkouts\stm32f1xx-hal-bead1e0eb6c95bf1\0f367e6\src\spi.rs:239:35
    |
239 |         mapr.modify_mapr(|_, w| w.spi3_remap().bit(REMAP::REMAP));
    |                                   ^^^^^^^^^^ help: there is an associated function with a similar name: `spi1_remap`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `stm32f1xx-hal`

To learn more, run the command again with --verbose.
```

I used new commit<0f367e6>, and it could not compile `stm32f1xx-hal`.
```
Checking stm32f1xx-hal v0.7.0 (https://github.com/stm32-rs/stm32f1xx-hal#0f367e60)
error[E0599]: no method named `spi3_remap` found for mutable reference `&mut stm32f1::W<u32, Reg<u32, _MAPR>>` in the current scope
   --> C:\Users\xxxxx\.cargo\git\checkouts\stm32f1xx-hal-bead1e0eb6c95bf1\0f367e6\src\spi.rs:239:35
    |
239 |         mapr.modify_mapr(|_, w| w.spi3_remap().bit(REMAP::REMAP));
    |                                   ^^^^^^^^^^ help: there is an associated function with a similar name: `spi1_remap`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `stm32f1xx-hal`
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -17,6 +17,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - Fix > 2 byte i2c reads
 - Send stop after acknowledge errors on i2c
 - Fix i2c interactions after errors
+- Fix SPI3 alternate function remapping.
 
 ### Changed
 - Use `cortex-m-rtic` instead of `cortex-m-rtfm` in the examples

--- a/src/spi.rs
+++ b/src/spi.rs
@@ -156,7 +156,7 @@ macro_rules! remap {
 remap!(Spi1NoRemap, SPI1, false, PA5, PA6, PA7);
 remap!(Spi1Remap, SPI1, true, PB3, PB4, PB5);
 remap!(Spi2NoRemap, SPI2, false, PB13, PB14, PB15);
-#[cfg(feature = "high")]
+#[cfg(any(feature = "high", feature = "connectivity"))]
 remap!(Spi3NoRemap, SPI3, false, PB3, PB4, PB5);
 #[cfg(feature = "connectivity")]
 remap!(Spi3Remap, SPI3, true, PC10, PC11, PC12);

--- a/src/spi.rs
+++ b/src/spi.rs
@@ -10,7 +10,7 @@
 
   - `SPI1` can use `(PA5, PA6, PA7)` or `(PB3, PB4, PB5)`.
   - `SPI2` can use `(PB13, PB14, PB15)`
-  - `SPI3` can use `(PB3, PB4, PB5)` or `(PC10, PC11, PC12)`
+  - `SPI3` can use `(PB3, PB4, PB5)` or only in connectivity line devices `(PC10, PC11, PC12)`
 
 
   ## Initialisation example
@@ -218,10 +218,35 @@ impl<REMAP, PINS> Spi<SPI3, REMAP, PINS, u8> {
     /**
       Constructs an SPI instance using SPI3 in 8bit dataframe mode.
 
+      The pin parameter tuple (sck, miso, mosi) should be `(PB3, PB4, PB5)` configured as `(Alternate<PushPull>, Input<Floating>, Alternate<PushPull>)`.
+
+      You can also use `NoSck`, `NoMiso` or `NoMosi` if you don't want to use the pins
+    */
+    #[cfg(not(feature = "connectivity"))]
+    pub fn spi3<F, POS>(
+        spi: SPI3,
+        pins: PINS,
+        mode: Mode,
+        freq: F,
+        clocks: Clocks,
+        apb: &mut APB1,
+    ) -> Self
+    where
+        F: Into<Hertz>,
+        REMAP: Remap<Periph = SPI3>,
+        PINS: Pins<REMAP, POS>,
+    {
+        Spi::<SPI3, _, _, u8>::_spi(spi, pins, mode, freq.into(), clocks, apb)
+    }
+
+    /**
+      Constructs an SPI instance using SPI3 in 8bit dataframe mode.
+
       The pin parameter tuple (sck, miso, mosi) should be `(PB3, PB4, PB5)` or `(PC10, PC11, PC12)` configured as `(Alternate<PushPull>, Input<Floating>, Alternate<PushPull>)`.
 
       You can also use `NoSck`, `NoMiso` or `NoMosi` if you don't want to use the pins
     */
+    #[cfg(feature = "connectivity")]
     pub fn spi3<F, POS>(
         spi: SPI3,
         pins: PINS,

